frontend_app_folder_advanced/src/container/FolderAdvanced.jsx

```diff
@@ -47,39 +51,62 @@ class FolderAdvanced extends React.Component {
     addAllResourceI18n(i18n, this.state.config.translation, this.state.loggedUser.lang)
     i18n.changeLanguage(this.state.loggedUser.lang)
 
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage },
+      { name: CUSTOM_EVENT.SHOW_APP(param.config.slug), handler: this.handleShowApp },
+      { name: CUSTOM_EVENT.HIDE_APP(param.config.slug), handler: this.handleHideApp },
+      { name: CUSTOM_EVENT.RELOAD_CONTENT(param.config.slug), handler: this.handleReloadContent },
+      { name: CUSTOM_EVENT.RELOAD_APP_FEATURE_DATA(param.config.slug), handler: this.handleReloadAppFeatureData }
+    ])
+
+    props.registerLiveMessageHandlerList([
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.MODIFIED, optionalSubType: TLM_ST.FOLDER, handler: this.handleFolderChanged },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.DELETED, optionalSubType: TLM_ST.FOLDER, handler: this.handleFolderChanged },
+      { entityType: TLM_ET.CONTENT, coreEntityType: TLM_CET.UNDELETED, optionalSubType: TLM_ST.FOLDER, handler: this.handleFolderChanged }
+    ])
   }
 
-  customEventReducer = ({ detail: { type, data } }) => {
+  static filterSubContentTypes (list) {
```

Comment:
do you really need it to be static ? 
since you only use it inside the class, you could use `this.` instead of `FolderAdvanced.`

Reply:
The advantage I see to use a static method is that it makes clear that it does not modify something on `this`… at the price of of less "elegant" way of calling it (which I suppose is what you point out). I could also define it as a function outside the class, what would be your opinion on this way?

Reply:
I asked myself this kind of question several times but couldn't find an answer that satisfy me completely.

I do like the fact, as you said, that is makes clear that it doesn't modify `this`
I don't mind the less elegant way of calling it

But it does bother me a little bit to use a specific keyword for something that it isn't intended to.

So I guess I would opt for the outside of class function